R/reporter-progress.R

```diff
@@ -57,6 +58,22 @@ ProgressReporter <- R6::R6Class("ProgressReporter",
       self$show_header()
     },
 
+    start_file = function(file) {
+      self$file_name <- file
+    },
+
+    start_test = function(context, test) {
+      if (is.null(context)) {
+
+        # We call the regular context() function rather than
+        # self$start_context() or self$.start_context() here because when this
+        # is called by a reporter inside the multi-reporter we need to assign
+        # self$.context in the multi-reporter rather than in the contained
+        # reporter.
+        context(self$file_name)
+      }
+    },
+
     start_context = function(context) {
```

Comment:
Should this all be moved to `start_file` or will you ignore `context()` calls in a different PR?

Reply:
I was going to do that / discuss it in a separate PR